mule-extensions-api/src/main/java/org/mule/extension/api/introspection/RuntimeComponentModel.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.api.introspection;
+
+import org.mule.extension.api.introspection.metadata.MetadataEnrichableModel;
+
+/**
+ * A specialization of the {@link ComponentModel} that represents an extension made component in their runtime
+ * specialization, like {@link RuntimeOperationModel} and {@link RuntimeSourceModel}, which adds
+ * behavioural components that are relevant to the extension's functioning when in runtime.
+ *
+ * @see ComponentModel
+ * @since 1.0
+ */
+public interface RuntimeComponentModel extends ComponentModel, InterceptableModel, ExceptionEnrichableModel, MetadataEnrichableModel
```

Comment:
It allows to work in the MetadataMediator in a polymorphic way, to be able to ask for their metadata and parameters with-out worry if the component is an Operation or a Source